test/parallel/test-fs-read-position-validation.mjs

```diff
@@ -13,89 +14,54 @@ const length = buffer.byteLength;
 
 // allowedErrors is an array of acceptable internal errors
 // For example, on some platforms read syscall might return -EFBIG
-async function testValid(position, allowedErrors = []) {
-  let fdSync;
-  try {
-    fdSync = fs.openSync(filepath, 'r');
-    fs.readSync(fdSync, buffer, offset, length, position);
-    fs.readSync(fdSync, buffer, { offset, length, position });
-  } catch (err) {
-    if (!allowedErrors.includes(err.code)) {
-      assert.fail(err);
-    }
-  } finally {
-    if (fdSync) fs.closeSync(fdSync);
-  }
+function testValidCbPositional([position, allowedErrors], callback) {
+  fs.open(filepath, 'r', common.mustSucceed((fd) => {
+    fs.read(fd, buffer, offset, length, position, common.mustCall((err) => {
+      fs.close(fd, common.mustSucceed(callback));
+      if (err && !allowedErrors?.includes(err.code)) {
+        assert.fail(err);
+      }
+    }));
```

Comment:
What do you think of that?
```suggestion
  fs.open(filepath, 'r', common.mustSucceed((fd) => {
    let alreadyCalled = false
    function handler(err) {
      if (err && !allowedErrors?.includes(err.code)) {
        fs.close(fd);
        assert.fail(err);
      } else if (alreadyCalled) {
        fs.close(fd, common.mustSucceed(callback));
      } else {
        alreadyCalled = true;
      }
    }
    fs.read(fd, buffer, offset, length, position, common.mustCall(handler));
    fs.read(fd, { buffer, offset, length, position }, common.mustCall(handler));
```

Reply:
Not so sequential, but should work just fine. This looks like an acceptance stage of callback hell. I like it. :)